### PR TITLE
Refactor: removed Result from FieldElement constructor

### PR DESCRIPTION
**Pull request by ajgara**
We apply modulo in the constructor instead.

### CODE REVIEW

File: src/math/field_element.rs [hunk 1]
```diff
@@ -67,7 +65,7 @@ impl ops::Add<FieldElement> for FieldElement {
     type Output = FieldElement;
 
     fn add(self, a_field_element: FieldElement) -> FieldElement {
-        FieldElement::new((self.value + a_field_element.value) % ORDER).unwrap()
```
Comment (MauroToscano) [inline src/math/field_element.rs:70]: % ORDER should remain in the operation

**Review by MauroToscano — COMMENTED** [review]
fe::one shouldn't be needed anymore,
same goes for fe::zero

File: src/math/field_element.rs (hunk 1 above)
Comment (schouhy) [reply src/math/field_element.rs:70]: I think it's fine without the % ORDER. It's the field element that corresponds to the class of the sum of the representatives.

Comment (MauroToscano) [reply src/math/field_element.rs:70]: > We apply modulo in the constructor instead.

Looks good then

**Comment by ajgara** [comment]
Already removed ::one and ::zero constructors. Had to add a neutral_element() method for CyclicGroup.

**Review by MauroToscano — APPROVED** [review]

**Pull request merged**